sormas-ui/src/main/java/de/symeda/sormas/ui/caze/messaging/SmsComponent.java

```diff
@@ -0,0 +1,88 @@
+package de.symeda.sormas.ui.caze.messaging;
+
+import com.vaadin.icons.VaadinIcons;
+import com.vaadin.shared.ui.ContentMode;
+import com.vaadin.ui.Alignment;
+import com.vaadin.ui.Component;
+import com.vaadin.ui.HorizontalLayout;
+import com.vaadin.ui.Label;
+import com.vaadin.ui.TextArea;
+import com.vaadin.ui.VerticalLayout;
+import com.vaadin.v7.ui.CustomField;
+
+import de.symeda.sormas.api.i18n.I18nProperties;
+import de.symeda.sormas.api.i18n.Strings;
+
+public class SmsComponent extends CustomField<String> {
+
+	private long missingPhoneNumbers;
+
+	private TextArea smsTextArea;
+
+	public SmsComponent(long missingPhoneNumbers) {
+		this.missingPhoneNumbers = missingPhoneNumbers;
+	}
+
+	@Override
+	protected Component initContent() {
+		final VerticalLayout mainLayout = new VerticalLayout();
+		mainLayout.setSpacing(false);
+		mainLayout.setSizeUndefined();
+		mainLayout.setWidth(100, Unit.PERCENTAGE);
+
+		if (missingPhoneNumbers > 0) {
+			mainLayout.addComponent(
+				new Label(
+					VaadinIcons.INFO_CIRCLE.getHtml() + " "
+						+ String.format(I18nProperties.getString(Strings.numberOfMissingPhoneNumbers), missingPhoneNumbers),
+					ContentMode.HTML));
+			mainLayout.addComponent(new Label());
+		}
+
+		mainLayout.addComponent(new Label(I18nProperties.getString(Strings.enterSMS)));
+
+		smsTextArea = new TextArea();
+		smsTextArea.setWidth(100, Unit.PERCENTAGE);
+		smsTextArea.setRows(4);
+		mainLayout.addComponent(smsTextArea);
+
+		final HorizontalLayout charactersLayout = new HorizontalLayout();
```

Comment:
Try to use the TextFieldWithMaxLengthWrapper that is used for the event description

Reply:
i also use this count to show the number of messages that is characters / 160 - should i extend this wrapper?

Reply:
Either that or do that independent of this component (e.g. by just adding another line below the field that counts the number of SMS); I leave that for you to decide